$NetBSD$

Add rpath so binaries can find liblightspark
Add addional libraries to liblightspark for successful linking

--- src/CMakeLists.txt.orig	2013-03-16 11:19:18.000000000 +0000
+++ src/CMakeLists.txt
@@ -140,7 +140,7 @@ ELSE (CMAKE_COMPILER_IS_GNUCC)
 ENDIF (CMAKE_COMPILER_IS_GNUCC)
 
 TARGET_LINK_LIBRARIES(spark ${CAIRO_LIBRARIES} ${ZLIB_LIBRARIES}
-	${Boost_LIBRARIES} ${LLVM_LIBS_CORE} ${LLVM_LIBS_JIT} ${LLVM_LDFLAGS}
+	${Boost_LIBRARIES} ${LLVM_LIBS_CORE} ${LLVM_LIBS_JIT} -lcompiler_rt -lc ${LLVM_LDFLAGS}
 	${OPTIONAL_LIBRARIES} ${GTK_LIBRARIES} ${FREETYPE_LIBRARIES} ${JPEG_LIBRARIES} ${PNG_LIBRARIES}
 	${PCRE_LIBRARIES} ${GLIBMM_LIBRARIES} ${GMODULE_LIBRARIES}
 	${XMLPP_LIBRARIES} ${CMAKE_DL_LIBS} ${EXTRA_LIBS_LIBRARIES}
@@ -161,6 +161,7 @@ INSTALL(TARGETS spark RUNTIME DESTINATIO
 # lightspark executable target
 IF(COMPILE_LIGHTSPARK)
   ADD_EXECUTABLE(lightspark main.cpp)
+  SET_TARGET_PROPERTIES(lightspark PROPERTIES LINK_FLAGS "-Wl,-rpath=${CMAKE_INSTALL_PREFIX}/lib/lightspark")
   TARGET_LINK_LIBRARIES(lightspark spark)
   #With STATICDEPS, all deps are compiled into spark
   IF(NOT STATICDEPS)
@@ -172,13 +173,14 @@ IF(COMPILE_LIGHTSPARK)
   INSTALL(FILES ${PROJECT_SOURCE_DIR}/src/lightspark.vert DESTINATION ${LSDATADIR})
   INSTALL(TARGETS lightspark RUNTIME DESTINATION ${BINDIR})
   IF(UNIX)
-    INSTALL(FILES ${PROJECT_SOURCE_DIR}/docs/man/lightspark.1 DESTINATION share/man/man1)
+    INSTALL(FILES ${PROJECT_SOURCE_DIR}/docs/man/lightspark.1 DESTINATION man/man1)
   ENDIF(UNIX)
 ENDIF(COMPILE_LIGHTSPARK)
 
 # tightspark executable target
 IF(COMPILE_TIGHTSPARK)
   ADD_EXECUTABLE(tightspark tightspark.cpp)
+  SET_TARGET_PROPERTIES(tightspark PROPERTIES LINK_FLAGS "-Wl,-rpath=${CMAKE_INSTALL_PREFIX}/lib/lightspark")
   TARGET_LINK_LIBRARIES(tightspark spark)
   #With STATICDEPS, all deps are compiled into spark
   IF(NOT STATICDEPS)
